infra/build/developer-tools/build/scripts/module-swapper/cmd/swap.go

```diff
@@ -0,0 +1,212 @@
+package cmd
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/pmezard/go-difflib/difflib"
+)
+
+type LocalTerraformModule struct {
+	Name      string
+	Dir       string
+	ModuleFQN string
+}
+
+var (
+	terraformExtension = "*.tf"
+	restoreMarker      = "[restore-marker]"
+	linebreak          = "\n"
+	localModules       = []LocalTerraformModule{}
+)
+
+// findSubModules generates slice of LocalTerraformModule for submodules
+func findSubModules(path, rootModuleFQN string) []LocalTerraformModule {
+	var subModules = make([]LocalTerraformModule, 0)
+	// if no modules dir, return empty slice
+	if err := validatePaths([]string{path}); err != nil {
+		log.Print("No submodules found")
+		return subModules
+	}
+	files, err := ioutil.ReadDir(path)
+	if err != nil {
+		log.Fatalf("Error finding submodules: %v", err)
+	}
+	absPath, err := filepath.Abs(path)
+	if err != nil {
+		log.Fatalf("Error finding submodule absolute path: %v", err)
+	}
+	for _, f := range files {
+		if f.IsDir() {
+			subModules = append(subModules, LocalTerraformModule{f.Name(), filepath.Join(absPath, f.Name()), fmt.Sprintf("%s//modules/%s", rootModuleFQN, f.Name())})
+		}
+	}
+	return subModules
+}
+
+// validatePaths validates a slice of paths
+func validatePaths(paths []string) error {
+	for _, p := range paths {
+		if _, err := os.Stat(p); err != nil {
+			return fmt.Errorf("Unable to find %s : %v", p, err)
+		}
+	}
+	return nil
+}
+
+// fmtTF uses the Terraform binary to format a tf file
```

Comment:
This is primarily to help with linting incase of any whitespace issues when we swap. In case terraform is not in path, this should not throw an error.

Reply:
Hmm, I'd rather make sure our swap doesn't break terraform fmt.

My reasoning is that swapping, then unswapping, should end up as a no-op. With this included, you would end up with formatted config.

Reply:
I have removed this and adjusted formatting in the utility. Now swapping and unswapping is no-op.